net/netns/netns_darwin.go

```diff
@@ -33,60 +35,154 @@ var bindToInterfaceByRouteEnv = envknob.RegisterBool("TS_BIND_TO_INTERFACE_BY_RO
 
 var errInterfaceStateInvalid = errors.New("interface state invalid")
 
-// controlLogf marks c as necessary to dial in a separate network namespace.
-//
-// It's intentionally the same signature as net.Dialer.Control
-// and net.ListenConfig.Control.
+// routeCache caches the results of interfaceIndexFor calls to avoid
+// spamming the AF_ROUTE socket. This is used for soft
+// isolation mode where we do many route lookups.
+type routeCacheEntry struct {
+	ifIndex int
+	err     error
+}
+
+var (
+	routeCache     *lru.Cache[string, routeCacheEntry]
+	routeCacheOnce sync.Once
+)
+
+func getRouteCache() *lru.Cache[string, routeCacheEntry] {
+	routeCacheOnce.Do(func() {
+		routeCache, _ = lru.New[string, routeCacheEntry](256)
```

Comment:
I think we should still add an expiry of 15s or something to this cache. I'm not confident that every possible routing or network change on the system will cause the net monitor to clear the cache.